src/function/gds/gds_utils.cpp

```diff
@@ -113,5 +119,23 @@ void GDSUtils::runVertexComputeIteration(processor::ExecutionContext* executionC
     }
 }
 
+void GDSUtils::runVertexComputeSparse(SparseFrontier& sparseFrontier, graph::Graph* graph,
```

Comment:
FYI, ideally even if the frontier is sparse, we should run it in parallel. You need a different "frontiermorselizer" implementation that can give morsels of frontiers to threads. This would require some rethinking of the frontiermorsel and frontiermorselizer interfaces, so it can work against both a sparse frontier and a dense frontier. But it would also avoid us having to implement a separate vertex compute or edge compute functions.

Reply:
Ditto opened an issue